src/unit_tests/analysisd/test_decoder_syscollector.c

```diff
@@ -1726,153 +3321,154 @@ void test_syscollector_dbsync_hardware_valid_msg_inserted(void **state)
     will_return(__wrap_wdbc_query_ex, 0);
 
     int ret = DecodeSyscollector(lf, &sock);
-
     assert_int_not_equal(ret, 0);
-}
-void test_syscollector_dbsync_os_valid_msg_inserted(void **state)
-{
-    Eventinfo *lf = *state;
 
-    const char *query = "agent 001 dbsync osinfo INSERTED "
-        "{"
-            "\"scan_time\":\"2021/10/29 14:26:24\","
-            "\"hostname\":\"93\","
-            "\"architecture\":\"94\","
-            "\"os_name\":\"95\","
-            "\"os_version\":\"96\","
-            "\"os_codename\":\"97\","
-            "\"os_major\":\"98\","
-            "\"os_minor\":\"99\","
-            "\"os_patch\":\"100\","
-            "\"os_build\":\"101\","
-            "\"os_platform\":\"102\","
-            "\"sysname\":\"103\","
-            "\"release\":\"104\","
-            "\"version\":\"105\","
-            "\"os_release\":\"106\","
-            "\"checksum\":\"107\","
-            "\"os_display_version\":\"108\","
-            "\"reference\":\"110\""
-        "}";
-    const char *result = "ok ";
-    int sock = 1;
+    // Clean previous package error, and exit
+    error_package = 1;
```

Comment:
We might set and clear the global variables with **setup** and **teardown** methods to avoid silently affecting other tests.

Reply:
Due to the number of functions defined, mainly because of the code coverage, and in the absence of other tools, it seems more appropriate to not continue adding functions excessively.